src/MapCodeExtractor.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace WP_CLI\I18n;
+
+use Gettext\Extractors\JsCode;
+use Gettext\Translations;
+use Peast\Syntax\Exception as PeastException;
+use WP_CLI;
+
+final class MapCodeExtractor extends JsCode {
+	use IterableCodeExtractor;
+
+	public static $options = [
+		'extractComments' => [ 'translators', 'Translators' ],
+		'constants'       => [],
+		'functions'       => [
+			'__'  => 'text_domain',
+			'_x'  => 'text_context_domain',
+			'_n'  => 'single_plural_number_domain',
+			'_nx' => 'single_plural_number_context_domain',
+		],
+	];
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public static function fromString( $string, Translations $translations, array $options = [] ) {
+		if ( ! array_key_exists( 'file', $options ) || substr( $options['file'], -7 ) !==  '.js.map' ) {
+			return;
+		}
+		$options['file'] = substr( $options['file'], 0, -7 ) . '.js';
+
+		try {
+			$options += static::$options;
+
+			$mapObject = json_decode( $string );
+
+			if ( ! isset( $mapObject->sourcesContent ) || ! is_array( $mapObject->sourcesContent ) ) {
+				return;
+			}
+
+			$string = implode( "\n", $mapObject->sourcesContent );
+
+			$functions = new JsFunctionsScanner( $string );
+
+			$functions->enableCommentsExtraction( $options['extractComments'] );
+			$functions->saveGettextFunctions( $translations, $options );
+		} catch ( PeastException $e ) {
+			WP_CLI::debug(
+				sprintf(
+					'Could not parse file %1$s: %2$s (line %3$d, column %4$d)',
+					$options['file'],
```

Comment:
This will point to the `.js` file, not the `.js.map` file, due to the change in line 31. That seems misleading in this case, as the source map failed to parse, not the `.js` file. The `.js` file would probably be a "webpack-ed" JS file, and the line and column references will be completely off.

I think the error message itself might need a clarification, as the line & column probably don't even exist in an actual file in that same location.